lib/services/doctor-service.ts

```diff
@@ -14,104 +12,32 @@ class DoctorService implements IDoctorService {
 	private static LinuxSetupDocsLink = "https://docs.nativescript.org/start/ns-setup-linux";
 
 	constructor(private $analyticsService: IAnalyticsService,
-		private $androidToolsInfo: IAndroidToolsInfo,
-		private $cocoapodsService: ICocoaPodsService,
 		private $hostInfo: IHostInfo,
 		private $logger: ILogger,
-		private $progressIndicator: IProgressIndicator,
-		private $staticConfig: IStaticConfig,
-		private $sysInfo: ISysInfo,
 		private $childProcess: IChildProcess,
-		private $config: IConfiguration,
-		private $npm: INodePackageManager,
 		private $opener: IOpener,
 		private $prompter: IPrompter,
-		private $fs: IFileSystem,
-		private $versionsService: IVersionsService,
-		private $xcprojService: IXcprojService) { }
+		private $versionsService: IVersionsService) { }
 
 	public async printWarnings(configOptions?: { trackResult: boolean }): Promise<boolean> {
-		let result = false;
-		const sysInfo = await this.$sysInfo.getSysInfo(this.$staticConfig.pathToPackageJson);
-
-		if (!sysInfo.adbVer) {
-			this.$logger.warn("WARNING: adb from the Android SDK is not installed or is not configured properly.");
-			this.$logger.out("For Android-related operations, the NativeScript CLI will use a built-in version of adb." + EOL
-				+ "To avoid possible issues with the native Android emulator, Genymotion or connected" + EOL
-				+ "Android devices, verify that you have installed the latest Android SDK and" + EOL
-				+ "its dependencies as described in http://developer.android.com/sdk/index.html#Requirements" + EOL);
-
-			this.printPackageManagerTip();
-			result = true;
-		}
-
-		if (!sysInfo.emulatorInstalled) {
-			this.$logger.warn("WARNING: The Android SDK is not installed or is not configured properly.");
-			this.$logger.out("You will not be able to build your projects for Android and run them in the native emulator." + EOL
-				+ "To be able to build for Android and run apps in the native emulator, verify that you have" + EOL
-				+ "installed the latest Android SDK and its dependencies as described in http://developer.android.com/sdk/index.html#Requirements" + EOL
-			);
+		const warnings = await doctor.getWarnings();
+		const hasWarnings = warnings.length > 0;
 
+		const hasAndroidWarnings = warnings.filter(warning => _.includes(warning.platforms, "Android")).length > 0;
```

Comment:
you can use `$devicePlatformsConstants.Android` here instead of hardcoding the string directly